### PR TITLE
Finetune against b60 nets + add ctl wrapper

### PR DESCRIPTION
**Pull request by ed1d1a8d**
Changes:
1. Reorders docker compose files to be quicker to build upon changes.
2. Add curriculum against b60-s7702m.
3. Create `ctl_job_run_wrapper.py` which can launch using either ctl or docker. We edit `launch-job.sh` to make use of this wrapper.
4. We edit `launch-job.sh` to support only launching extra victimplay jobs.

### CODE REVIEW

File: kubernetes/ctl_job_run_wrapper.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+            for _ in range(job_replicas):
+                if job_gpus == 0:
+                    gpu_indices.append([])
+                elif collo_allowed == 1:
+                    assert job_gpus == 1
+                    assert job_replicas == 1
+                    if collo_idx == -1:
+                        collo_idx = min_unused_idx
+                    gpu_indices.append([collo_idx])
+                    min_unused_idx += 1
```
Comment (tomtseng) [inline kubernetes/ctl_job_run_wrapper.py:111]: move this line inside `if collo_idx == -1`?

File: kubernetes/launch-job.sh [hunk 2]
```diff
@@ -187,8 +209,46 @@ else
   CURRICULUM_CMD="/go_attack/kubernetes/curriculum.sh $RUN_NAME $VOLUME_NAME $CURRICULUM"
 fi
 
+if [ -n "${ONLY_LAUNCH_CURRICULUM:-}" ]; then
```
Comment (tomtseng) [inline kubernetes/launch-job.sh:212]: is this mainly for local (non-hofvarpnir) runs? on hofvarpnir you'd have to kill the old curriculum, but i'm not certain if there's a way to kill just the curriculum pod without killing the whole `go-train-"$1"-vital` job

File: kubernetes/launch-job.sh [hunk 3]
```diff
@@ -187,8 +209,46 @@ else
...
+      --command "$VICTIMPLAY_CMD" \
+      --gpu 1 \
+      --name "$EXTRA_VICTIMPLAY_JOB_NAME" \
+      --replicas "${EXTRA_VICTIMPLAY_GPUS}" \
+      $EXTRA_VICTIMPLAY_HP_FLAG
+
+  if [ $MIN_VICTIMPLAY_GPUS -eq 0 ]; then
+    echo "Running extra victimplay jobs only."
+    exit 0
+  fi
```
Comment (tomtseng) [inline kubernetes/launch-job.sh:247]: i feel like this is slightly weird because I might expect setting `MIN_VICTIMPLAY_GPUS` to still launch evaluate_loop, train, shuffle_and_export, curriculum, and gating.
not a huge deal, you've kinda documented this behavior already under `-g GPUS, --victimplay-gpus GPUS`

File: configs/compute/1gpu.cfg [hunk 4]
```diff
@@ -1,11 +1,11 @@
-numGameThreads = 256
+numGameThreads = 128
 
 # GPU Settings-------------------------------------------------------------------------------
 # 1xA6000
 # 2 sockets of AMD EPYC 7763 64-Core Processors = 128 physical cores
 #                                                 each core has 2 threads
 
-nnMaxBatchSize = 256
+nnMaxBatchSize = 128
```
[the remark this answers is not in the document]
Comment (ed1d1a8d) [reply configs/compute/1gpu.cfg:8]: I think so. The two gpus with the lowest memory are running with batch size 128 here.
image

[the remark this answers is not in the document]
Comment (ed1d1a8d) [reply configs/compute/1gpu.cfg]: Adjusted to match https://github.com/AlignmentResearch/go_attack/commit/9b7445e4320cafa621012717929544ec75ebb2cb

File: kubernetes/ctl_job_run_wrapper.py (hunk 1 above)
Comment (ed1d1a8d) [reply kubernetes/ctl_job_run_wrapper.py:111]: Good catch.

File: configs/active-experiment.cfg [hunk 5]
```diff
@@ -1,9 +1,8 @@
 # Must be combined with a config file from the compute/ folder to work
 @include amcts/victimplay.cfg
 
-maxVisits0 = 32 # increase this to make victim stronger
-# Uncomment this for pass-hardening
-# passingBehavior0 = avoid-pass-alive-territory
+# Idx 0 is victim
+passingBehavior0 = avoid-pass-alive-territory
```
[the remark this answers is not in the document]
Comment (ed1d1a8d) [reply configs/active-experiment.cfg:5]: Reverted this active-experiment config to match the main branch.

File: kubernetes/launch-job.sh (hunk 3 above)
Comment (ed1d1a8d) [reply kubernetes/launch-job.sh:247]: Happy to change this is you have a better suggestion for how to only launch extra jobs. I just did this cause it was the easiest thing to do.

File: kubernetes/launch-job.sh (hunk 2 above)
Comment (ed1d1a8d) [reply kubernetes/launch-job.sh:212]: Hmm, possibly. This was a while back and I forget if I used this on hofvarpnir or not. Do you know how you would in principle hot-swap the curriculum on hofvarpnir?

File: compose/cpp/Dockerfile [hunk 6]
```diff
@@ -136,23 +136,30 @@ COPY README.md /go_attack/README.md
...
 # Copy over other files
-COPY ./kubernetes/ /go_attack/kubernetes
 COPY ./scripts/ /go_attack/scripts
-
 COPY ./engines/KataGo-custom/cpp/evaluate_loop.sh /engines/KataGo-custom/cpp/evaluate_loop.sh
 COPY ./engines/KataGo-raw/cpp/configs /engines/KataGo-raw/cpp/configs
 
+# Set entrypoint
+COPY ./compose/cpp/entrypoint.sh entrypoint.sh
+RUN chmod +x entrypoint.sh
```
[the remark this answers is not in the document]
Comment (ed1d1a8d) [reply compose/cpp/Dockerfile]: Done.
